lib/matplotlib/projections/polar.py

```diff
@@ -20,39 +20,50 @@ class PolarTransform(mtransforms.Transform):
     The base polar transform.
 
     This transform maps polar coordinates ``(theta, r)`` into Cartesian
-    coordinates ``(x, y) = (r * cos(theta), r * sin(theta))`` (but does not
+    coordinates ``(x, y) = (r * cos(theta), r * sin(theta))``
+    (but does not fully transform into Axes coordinates or
     handle positioning in screen space).
 
+    This transformation is designed to be applied to data after any scaling
+    along the radial axis (e.g. log-scaling) has been applied to the input
+    data.
+
     Path segments at a fixed radius are automatically transformed to circular
     arcs as long as ``path._interpolation_steps > 1``.
     """
 
     input_dims = output_dims = 2
 
     def __init__(self, axis=None, use_rmin=True,
-                 _apply_theta_transforms=True):
+                 _apply_theta_transforms=True, *, scale_transform=None):
         super().__init__()
         self._axis = axis
         self._use_rmin = use_rmin
         self._apply_theta_transforms = _apply_theta_transforms
+        self._scale_transform = scale_transform
 
     __str__ = mtransforms._make_str_method(
         "_axis",
         use_rmin="_use_rmin",
         _apply_theta_transforms="_apply_theta_transforms")
 
+    def _get_rorigin(self):
+        # Get lower r limit after being scaled by the radial scale transform
+        return self._scale_transform.transform(
+            (0, self._axis.get_rorigin()))[1]
+
     def transform_non_affine(self, tr):
         # docstring inherited
-        t, r = np.transpose(tr)
+        theta, r = np.transpose(tr)
         # PolarAxes does not use the theta transforms here, but apply them for
         # backwards-compatibility if not being used by it.
         if self._apply_theta_transforms and self._axis is not None:
-            t *= self._axis.get_theta_direction()
-            t += self._axis.get_theta_offset()
+            theta *= self._axis.get_theta_direction()
+            theta += self._axis.get_theta_offset()
```

Comment:
Is it possible to add a test that uses these lines?

Reply:
It looks like `_apply_theta_transforms=True` case is not used internally, but as a backwards compatability wrapper for users who used `PolarTransform` before its behaviour was changed. I think adding a meaninful test would be tricky here.